### PR TITLE
adds content-security-policy to upgrade requests

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ const pluginDefaults = {
     'default-src': ['https:'],
     'report-uri': 'http://localhost/csp_reports'
   },
-  headerKey: 'Content-Security-Policy-Report-Only'
+  headerKey: 'Content-Security-Policy-Report-Only',
+  policyHeaderKey: 'Content-Security-Policy',
```
Comment (jgallen23) [inline index.js]: lets make this dummy proof and have the option be `upgradeInsecureRequests: true`

File: test/test.mixed.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ lab.afterEach((done) => {
 });
 
 
-lab.test('should add default header to incoming requests of the indicated variety ', (allDone) => {
+lab.test('should add default headers to incoming requests of the indicated variety ', (allDone) => {
```
Comment (jgallen23) [inline test/test.mixed.js:24]: we need a test if its false

File: index.js [hunk 3]
```diff
@@ -8,9 +8,13 @@ const pluginDefaults = {
     'default-src': ['https:', 'unsafe-inline', 'unsafe-eval'],
     'report-uri': '/csp_reports'
   },
-  headerKey: 'Content-Security-Policy-Report-Only'
+  headerKey: 'Content-Security-Policy-Report-Only',
```
Comment (jgallen23) [inline index.js]: should we change this to reportOnly: true? and if it's false, then have the key be `content-security-policy`?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (orthagonal) [reply index.js]: I feel like it makes more sense if we have 'reportErrors' and 'upgradeInsecureRequests', and they can just set either or both of them to false.

**Pull request merged**